http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ListenerInvokingChannelPool.java

```diff
@@ -84,7 +85,7 @@ public ListenerInvokingChannelPool(Supplier<Promise<Channel>> promiseFactory,
                                        List<ChannelPoolListener> listeners) {
         this.delegatePool = delegatePool;
         this.promiseFactory = promiseFactory;
-        this.listeners = listeners;
+        this.listeners = Collections.unmodifiableList(listeners);
```

Comment:
Same here, seems a bit overkill for internal classes.

Reply:
removing for now , will need to comeup with a generic supress rule for internal classes